### PR TITLE
[DOCS] Move HA section to the main TOC

### PR DESCRIPTION
**Pull request by ologvinova**
Reason: the section relates not only to the standalone but to the Cluster API use case.

Signed-off-by: ologvinova <ologvinova@mirantis.com>

### CODE REVIEW

File: docs/ha.md [hunk 1]
```diff
@@ -1,77 +1,97 @@
-# Highly available controlplanes
+# Highly available control planes
```
Comment (makhov) [inline docs/ha.md:1]: WDYT, does it make sense to emphasize in the title that this is for hosted control planes?

**Review by makhov — APPROVED** [review]

**Comment by dzhigzhitzhapov** [comment]
LGTM, Magda's comments are quite interesting, though.

File: docs/ha.md [hunk 2]
```diff
@@ -1,77 +1,97 @@
...
+!!! note
 
-The control planes managed by `k0smotron` are k0s control planes. As k0s comes with support for using SQL DBs as data store (via Kine) you can use HA databases instead of Etcd. This enables you to use e.g. Postgres operator, MySQL operator or cloud provider managed databases as the data store for the control planes.
+   Highly available control planes are supported for the standalone and
+   Cluster API in-cluster use cases.
+
+As the nature of Kubernetes workloads, in our case the cluster control planes,
+is quite dynamic, it poses a challenge to set up highly available etcd cluster
+for the control plane. In k0smotron, we are solving the challenge by
+"externalizing" the HA setup of data storage for the control plane.
```
[the remark this answers is not in the document]
Comment (ologvinova) [reply docs/ha.md:11]: Thanks! Rephrased.

**Review by MagdaDziadosz — APPROVED** [review]

File: docs/ha.md [hunk 3]
```diff
@@ -1,77 +1,97 @@
...
-    type: LoadBalancer
-  kineDataSourceSecretName: database-credentials
-EOF
-```
-
-**Note**: The secret must be in the same namespace as the cluster and the key must be `K0SMOTRON_KINE_DATASOURCE_URL`.
+In this example, we provide instruction on how to use
+[Postgres operator](https://postgres-operator.readthedocs.io/en/latest/)
+to manage the data store of a control plane. Use this instruction as an example
+for the required data store resource.
```
[the remark this answers is not in the document]
Comment (ologvinova) [reply docs/ha.md]: Sounds nicer, thanks! Rephrased.

**Pull request merged**